radam.py

```diff
@@ -15,8 +15,11 @@ def __init__(self, params, lr=1e-3, betas=(0.9, 0.999), eps=1e-8, weight_decay=0
             raise ValueError("Invalid beta parameter at index 1: {}".format(betas[1]))
         
         self.degenerated_to_sgd = degenerated_to_sgd
-        defaults = dict(lr=lr, betas=betas, eps=eps, weight_decay=weight_decay)
-        self.buffer = [[None, None, None] for ind in range(10)]
+        if isinstance(params, (list, tuple)) and len(params) > 0 and isinstance(params[0], dict):
```

Comment:
Wondering why it will be necessary to manually set the `buffer` field for params. In other words, with line 22 setting the default value of buffer, why we need line 18-21?

Reply:
defaults shares object not value. Without 18-21, buffer is shared.
These lines means that 'if user sets multiple params groups that have different betas from default, params group has seperate buffer'

Reply:
thanks for explaining, that makes sense